File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utils/CancelableAction.java
package org.firstinspires.ftc.teamcode.utils;

import com.acmerobotics.roadrunner.Action;
/**
 * Extends the Action interface to include a cancel method.
 */
public interface CancelableAction extends Action {
    /**
     * Cancels the action.
     */
    void cancel();
}

File: TeamCode/src/main/java/org/firstinspires/ftc/utils/CancelableAction.kt
package org.firstinspires.ftc.utils

import com.acmerobotics.roadrunner.Action

interface CancelableAction : Action {
    /**
     * Cancels the action.
     * Implementations should stop any ongoing actions and reset any state.
     * Implements RoadRunner's [Action] interface.
     * @see Action.run()
     */
    fun cancel()
}